src/coreclr/src/gc/gc.cpp

```diff
@@ -25724,33 +27830,10 @@ void gc_heap::reloc_ref_in_shortened_obj (uint8_t** address_to_set_card, uint8_t
     relocate_address (address_to_reloc THREAD_NUMBER_ARG);
     if (address_to_reloc)
     {
-        dprintf (3, ("SR %Ix: %Ix->%Ix", (uint8_t*)address_to_reloc, old_val, *address_to_reloc));
-    }
-
-    //check_demotion_helper (current_saved_info_to_relocate, (uint8_t*)pval);
-    uint8_t* relocated_addr = *address_to_reloc;
-    if ((relocated_addr < demotion_high) &&
-        (relocated_addr >= demotion_low))
-    {
-        dprintf (3, ("set card for location %Ix(%Ix)",
-                    (size_t)address_to_set_card, card_of((uint8_t*)address_to_set_card)));
-
-        set_card (card_of ((uint8_t*)address_to_set_card));
+        dprintf (3, ("SR %Ix: %Ix->%Ix", (uint8_t*)address_to_reloc, old_val, *address_to_reloc));        
     }
-#ifdef MULTIPLE_HEAPS
-    else if (settings.demotion)
-    {
-        gc_heap* hp = heap_of (relocated_addr);
-        if ((relocated_addr < hp->demotion_high) &&
-            (relocated_addr >= hp->demotion_low))
-        {
-            dprintf (3, ("%Ix on h%d, set card for location %Ix(%Ix)",
-                        relocated_addr, hp->heap_number, (size_t)address_to_set_card, card_of((uint8_t*)address_to_set_card)));
 
-            set_card (card_of ((uint8_t*)address_to_set_card));
-        }
-    }
-#endif //MULTIPLE_HEAPS
+    check_demotion_helper (address_to_reloc, (uint8_t*)address_to_set_card);
```

Comment:
So this is just calling the helper instead of repeating the code here?

Reply:
yes, I think when I wrote the original code I had something else going that made it not exactly the same. but that difference was gone so I should have converted it to calling the method.

---
In reply to: [534273646](https://github.com/dotnet/runtime/pull/45172#discussion_r534273646) [](ancestors = 534273646)